contracts/modules/v2/StakingPool.sol

```diff
@@ -617,8 +673,8 @@ contract StakingPool is IStakingPool, ERC721 {
 
 
     // Move the user's shares from the initial deposit to the new one.
-    newDeposit.rewardsShares += initialDeposit.rewardsShares;
-    newDeposit.stakeShares += initialDeposit.stakeShares;
+    newDeposit.rewardsShares += initialDeposit.rewardsShares + newRewardsShares;
```

Comment:
isn't this supposed to be:

```newDeposit.rewardsShares += newRewardsShares;````

the += is confusing.

if ```newDeposit.rewardsShares``` is zero i'd just do +=

Reply:
The new deposit can be a deposit that already exists which is topped up with the new rewards shares and the reward shares from the initial deposit which is extended.

Reply:
Would renaming `new` to `target` dispell the confusion and make it more obvious that the deposit is not guaranteed to be 0?

Reply:
```The new deposit can be a deposit that already exists which is topped up with the new rewards shares and the reward shares from the initial deposit which is extended.```

how about a comment on this. this makes sense